consensus/e2e_tests/state_sync_test.go

```diff
@@ -179,3 +181,130 @@ func TestStateSync_ServerGetBlock_FailNonExistingBlock(t *testing.T) {
 	_, err = WaitForNetworkStateSyncEvents(t, clockMock, eventsChannel, errMsg, numExpectedMsgs, 250, false)
 	require.Error(t, err)
 }
+
+func TestStateSync_UnsynchedPeerSynchs_Success(t *testing.T) {
```

Comment:
Even if you don't add more tests, think of other tests you want to implement and add headers for them. I can try to think of a few once you take another stab at it.

For example, having 4 nodes, all at different heights, and validating that they all catch up

Reply:
I added few more test, I may update again.